include/net/nrf_cloud_coap.h

```diff
@@ -0,0 +1,253 @@
+ * @retval -EAGAIN if an error occurred while loading the connection.
+ * @retval -EINVAL if the operation could not be performed.
+ * @retval 0 If successful.
+ */
+int nrf_cloud_coap_resume(void);
+
+/**
+ * @brief Disconnect the nRF Cloud CoAP connection
+ *
+ * This does not teardown the thread in coap_client, as there is no way to do so.
+ * The thread's call to poll(sock) will fail, resulting in an error message.
+ * This is expected. Call nrf_cloud_coap_connect() to re-establish the connection, and
+ * the thread in coap_client will resume.
+ *
+ * @return 0 if the socket was closed successfully, or a negative error number.
+ */
+int nrf_cloud_coap_disconnect(void);
+
+/* nRF Cloud service functions */
+
+/**
+ * @brief nRF Cloud CoAP Assisted GPS (A-GPS) data request.
+ *
+ * @param[in]     request Data to be provided in API call.
+ * @param[in,out] result Structure pointing to caller-provided buffer in which to store A-GPS data.
+ *
+ *  @retval -EINVAL will be returned, and an error message printed, if invalid parameters
+ *          are given.
+ *  @retval -ENOENT will be returned if there was no A-GPS data requested for the specified
+ *          request type.
+ *  @retval -ENOBUFS will be returned, and an error message printed, if there is not enough
+ *          buffer space to store retrieved AGPS data.
+ * @retval 0 If successful.
+ */
+int nrf_cloud_coap_agps_data_get(struct nrf_cloud_rest_agps_request const *const request,
+				 struct nrf_cloud_rest_agps_result *result);
+
+/**
+ * @brief nRF Cloud Predicted GPS (P-GPS) request URL.
+ *
+ *  After a successful call to this function, pass the file_location to
+ *  nrf_cloud_pgps_update(), which then downloads and processes the file's binary P-GPS data.
+ *
+ * @param[in]     request       Data to be provided in API call.
+ * @param[in,out] file_location Structure that will contain the host and path to
+ *                              the prediction file.
+ *
+ * @retval 0 If successful.
+ *          Otherwise, a (negative) error code is returned.
+ */
+int nrf_cloud_coap_pgps_url_get(struct nrf_cloud_rest_pgps_request const *const request,
+				 struct nrf_cloud_pgps_result *file_location);
+
+/**
+ * @brief Send a sensor value to nRF Cloud.
+ *
+ *  The CoAP message is sent as a non-confirmable CoAP message.
+ *
+ * @param[in]     app_id The app_id identifying the type of data. See the values in nrf_cloud_defs.h
+ *                       that begin with  NRF_CLOUD_JSON_APPID_. You may also use custom names.
+ * @param[in]     value  Sensor reading.
+ *
+ * @retval 0 If successful.
+ *          Otherwise, a (negative) error code is returned.
+ */
+int nrf_cloud_coap_sensor_send(const char *app_id, double value);
+
+/**
+ * @brief Send the device location in the :ref:`nrf_cloud_gnss_data` PVT field to nRF Cloud.
+ *
+ *  The CoAP message is sent as a non-confirmable CoAP message. Only
+ *  :ref:`NRF_CLOUD_GNSS_TYPE_PVT` is supported.
```

Comment:
I'm afraid the :ref: links within doxygen docs do not work: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/documentation/styleguide.html#references